staging/src/k8s.io/apiextensions-apiserver/hack/update-codegen.sh

```diff
@@ -28,11 +28,17 @@ kube::codegen::gen_helpers \
     --output-base "$(dirname "${BASH_SOURCE[0]}")/../../.." \
     --boilerplate "${SCRIPT_ROOT}/hack/boilerplate.go.txt"
 
+if [[ "${UPDATE_API_KNOWN_VIOLATIONS:-}" == "true" ]]; then
+    update_report="--update-report"
+fi
+
 kube::codegen::gen_openapi \
     --input-pkg-root k8s.io/apiextensions-apiserver/pkg \
     --extra-pkgs k8s.io/api/autoscaling/v1 `# needed for Scale type` \
     --output-pkg-root k8s.io/apiextensions-apiserver/pkg/generated \
     --output-base "$(dirname "${BASH_SOURCE[0]}")/../../.." \
+    --report-filename "${SCRIPT_ROOT}/api_violation_exceptions.list" \
```

Comment:
I like the partitioning of the generation a lot

Not sure about locating the exceptions file at the root of the module, for a couple reasons:
* it's ugly (would be on the main page of https://github.com/kubernetes/apiextensions-apiserver, for example)
* the current location routes to API review due to https://github.com/kubernetes/kubernetes/tree/master/api, which we'd lose

is there an API package we could tuck it away into?

Reply:
The primary requirement is that it has to run from the root of each module (standalone) and from k/k.

Options:

1) Put it in a subdir of each module
2) Put it in a dotfile of each module
3) pass `--report-filename` as an arg to each module's update-codegen.sh
4) use a hidden env var to pass the moral equivalent of `--report-filename` to each module
5) try to sniff up the directory tree and figure out if we are running from k/k or not

(1) and (2) are OK to me.  The rest all kind of stink.  We simulataneously want these repos to work as standalone things AND we want them to dump their report into a parent which might not exist.

If I am squeezed, I will suggest #3, but I'll have to hold me nose while I do it :)

We could do #1 with OWNERS ?

Reply:
1 seems ok and works with OWNERS

4 actually doesn't bother me as a tweak to the `UPDATE_API_KNOWN_VIOLATIONS` envvar we're already passing down to trigger this... is it worse to trigger off a non-empty `UPDATE_API_KNOWN_VIOLATIONS_FILE` envvar instead and require that to either be relative to $PWD or an absolute path?

Reply:
> How about ...

any of those seem fine

> I don't really know what we do with these other than look at them and then approve anyway :)

We really do avoid growing them for new stuff, and it regularly catches name mismatches or failure to explicitly pick a list type

Reply:
Or I can just use the old names

Reply:
new commit pushed - take a look